### ISSUE TITLE
--features=extra Build fails with "error: failed to run custom build command for `memchr v2.4.1`"

### ISSUE THREAD

**Issue by yingzhu146** [nushell/nushell#5330]
### Describe the bug

I'm trying to get access to the query plugins etc and thus compile with --features=extra.

Without it it compiles fine, additionally the brew provided version works fine as well.

verbose output of `cargo build --workspace --features=extra`/ or `cargo install --all-features --path . --verbose`
<details>
  <summary>

```bash
error: failed to run custom build command for `libc v0.2.124`

Caused by:
  process didn't exit successfully: `/Users/yingzhu/nushell/target/release/build/libc-45d80016f98f0c38/build-script-build` (signal: 9, SIGKILL: kill)
warning: build failed, waiting for other jobs to finish...
error: failed to run custom build command for `syn v1.0.91`

Caused by:
  process didn't exit successfully: `/Users/yingzhu/nushell/target/release/build/syn-89b5b75cdb61b168/build-script-build` (signal: 9, SIGKILL: kill)
error: failed to run custom build command for `libc v0.2.124`

Caused by:
  process didn't exit successfully: `/Users/yingzhu/nushell/target/release/build/libc-45d80016f98f0c38/build-script-build` (signal: 9, SIGKILL: kill)
error: failed to run custom build command for `serde v1.0.136`

Caused by:
  process didn't exit successfully: `/Users/yingzhu/nushell/target/release/build/serde-039252e623f6cb0c/build-script-build` (signal: 9, SIGKILL: kill)
     Running `/Users/yingzhu/nushell/target/release/build/memchr-8fc659c2a86e90ad/build-script-build`
error: failed to run custom build command for `proc-macro2 v1.0.37`

Caused by:
  process didn't exit successfully: `/Users/yingzhu/nushell/target/release/build/proc-macro2-61a5055c140e61f2/build-script-build` (signal: 9, SIGKILL: kill)
error: failed to run custom build command for `memchr v2.4.1`

Caused by:
  process didn't exit successfully: `/Users/yingzhu/nushell/target/release/build/memchr-8fc659c2a86e90ad/build-script-build` (signal: 9, SIGKILL: kill)
error: failed to run custom build command for `memchr v2.4.1`

Caused by:
  process didn't exit successfully: `/Users/yingzhu/nushell/target/release/build/memchr-8fc659c2a86e90ad/build-script-build` (signal: 9, SIGKILL: kill)
error: failed to compile `nu v0.61.1 (/Users/yingzhu/nushell)`, intermediate artifacts can be found at `/Users/yingzhu/nushell/target`
```
when running
```   Compiling serde v1.0.136
     Running `/Users/yingzhu/nushell/target/release/build/libc-45d80016f98f0c38/build-script-build`
     Running `/Users/yingzhu/nushell/target/release/build/libc-45d80016f98f0c38/build-script-build`
     Running `/Users/yingzhu/nushell/target/release/build/syn-89b5b75cdb61b168/build-script-build`
     Running `/Users/yingzhu/nushell/target/release/build/proc-macro2-61a5055c140e61f2/build-script-build`
     Running `/Users/yingzhu/nushell/target/release/build/memchr-8fc659c2a86e90ad/build-script-build`
     Running `/Users/yingzhu/nushell/target/release/build/serde-039252e623f6cb0c/build-script-build`
```
</summary>
</details>

### How to reproduce

run `cargo install --all-features --path . --verbose`

### Expected behavior

install

### Configuration

(note this is the installation without --features=extra)
| key                | value                                       |
| ------------------ | ------------------------------------------- |
| version            | 0.61.1                                      |
| branch             | main                                        |
| tag                |                                             |
| short_commit       | 187f2454                                    |
| commit_hash        | 187f2454c8500d91b703fb72ca10c6b3ab84b1d0    |
| commit_date        | 2022-04-25 23:44:57 +00:00                  |
| build_os           | macos-aarch64                               |
| rust_version       | rustc 1.62.0-nightly (60e50fc1c 2022-04-04) |
| rust_channel       | nightly-aarch64-apple-darwin                |
| cargo_version      | cargo 1.62.0-nightly (1ef1e0a12 2022-03-31) |
| pkg_version        | 0.61.1                                      |
| build_time         | 2022-04-25 19:53:05 -04:00                  |
| build_rust_channel | release                                     |
| features           | default, trash, which, zip                  |
| installed_plugins  |                                             |

### Additional context

None except that y'all are awesome and Nushell is amazing

**Comment by jntrnr** [comment]
@yingzhu146 - can you try building with the latest Rust stable compiler?

**Comment by yingzhu146** [comment]
That was quick, thanks for the reply @jntrnr! Sure, nuked the repo, redownloaded, then  - `cargo +stable install --path . --`'d, however same story it seems
<details>
  <summary>
```shell
   Compiling num-integer v0.1.44
     Running `/Users/yingzhu/nushell/target/release/build/proc-macro2-e6c516e36bd529a5/build-script-build`
     Running `/Users/yingzhu/nushell/target/release/build/memchr-6ddaf6f0f3017978/build-script-build`
     Running `/Users/yingzhu/nushell/target/release/build/serde-757def6e8a3ae772/build-script-build`
     Running `rustc --crate-name build_script_build /Users/yingzhu/.cargo/registry/src/github.com-1ecc6299db9ec823/num-integer-0.1.44/build.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debug-assertions=off --cfg 'feature="i128"' --cfg 'feature="std"' -C metadata=203529c2f5a20c4f -C extra-filename=-203529c2f5a20c4f --out-dir /Users/yingzhu/nushell/target/release/build/num-integer-203529c2f5a20c4f -C strip=debuginfo -L dependency=/Users/yingzhu/nushell/target/release/deps --extern autocfg=/Users/yingzhu/nushell/target/release/deps/libautocfg-003fe45f7f2df9d9.rlib --cap-lints allow`
     Running `rustc --crate-name build_script_build /Users/yingzhu/.cargo/registry/src/github.com-1ecc6299db9ec823/num-traits-0.2.14/build.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debug-assertions=off --cfg 'feature="default"' --cfg 'feature="i128"' --cfg 'feature="libm"' --cfg 'feature="std"' -C metadata=f17191737746885c -C extra-filename=-f17191737746885c --out-dir /Users/yingzhu/nushell/target/release/build/num-traits-f17191737746885c -C strip=debuginfo -L dependency=/Users/yingzhu/nushell/target/release/deps --extern autocfg=/Users/yingzhu/nushell/target/release/deps/libautocfg-003fe45f7f2df9d9.rlib --cap-lints allow`
error: failed to run custom build command for `libc v0.2.124`

Caused by:
  process didn't exit successfully: `/Users/yingzhu/nushell/target/release/build/libc-6671a8e3a9514ec5/build-script-build` (signal: 9, SIGKILL: kill)
warning: build failed, waiting for other jobs to finish...
error: failed to compile `nu v0.61.1 (/Users/yingzhu/nushell)`, intermediate artifacts can be found at `/Users/yingzhu/nushell/target`

```
</summary>
</details>


Oddly enough, I now can't reproducibly build the stable version for anything either 🤔

**Comment by fdncred** [comment]
I build on my Mac M1 on stable every day. This is the current rust version I'm using. Is this the same version you're building with on stable? `rustc 1.60.0 (7737e0b5c 2022-04-04)`

**Comment by yingzhu146** [comment]
Thanks a lot @fdncred - yes, same version

```bash
rustc --version
rustc 1.60.0 (7737e0b5c 2022-04-04)
```
also just for completeness
```
rustup toolchain list                                                                                                                                          
stable-aarch64-apple-darwin (default)
nightly-aarch64-apple-darwin
```

I tried `install-all.sh` - same issue. Also just tried turning it on and off again, because, you know, you never know! (: persists over restarts. A bit out of ideas here. I'll try to update on the most recent osx and see what happens tomorrow.

Really appreciate both of your help here, let me know if there's other stuff I could try/provide

**Comment by jntrnr** [comment]
@yingzhu146 - I looked back over this. I'm not sure why, but something is making me wonder if this is maybe related to a C compiler on your system?

Either that, or something is causing build.rs files to fail to run. Maybe an environment variable that rustc or cargo uses is set to something it doesn't expect.

**Comment by yingzhu146** [comment]
Unfortunately update didn't help.

So these are both great suggestion, I have a very similar hunch @jntrnr - I do a lot of scientific computing unfortunately work with a ton of C/C++ with python bindings that I had to hand-compile, which leads to me having unfortunately 3 different systems which all expect to be the main dependency manager (conda, brew, and sytem binaries) - so e.g. I have 3 different cargos (one system wide, one from a conda env and one from brew) and often these build scripts take the wrong paths (and sometimes I even have to do funny stuff like overwrite CMakeLists etc so they don't misbehave). Let me try to nuke my env vars - in the mean time any env vars for your build process that you think might be relevant/important?

**Comment by fdncred** [comment]
For me, there are no specific env var that I set manually for a build.

**Comment by yingzhu146** [comment]
WOHO.

Right on the money @jntrnr, was indeed an env var.

for the record, here's how I debugged, which was actually quite nice using nu.

1) figure out all env vars which touch dynamic lib search paths
```nushell
env | find "lib"
```
2) remove the ones that looked to pull dylibs across environments
in my case I had modified these out of desperation in earlier compilation odysees
```
export CC
export LD_LIBRARY_PATH
export CPATH
export LIBRARY_PATH
```
3) start new shell, profit

Really appreciated the prompt and kind help from both of you!
